packages/eslint-plugin/src/util/collectUnusedVariables.ts

```diff
@@ -423,9 +423,7 @@ function isMergableExported(variable: TSESLint.Scope.Variable): boolean {
  * @returns True if the variable is exported, false if not.
  */
 function isExported(variable: TSESLint.Scope.Variable): boolean {
-  const definition = variable.defs[0];
-
-  if (definition) {
+  const exportedDefinition = variable.defs.find(definition => {
```

Comment:
[Nitpick] This can be ever so slightly simplified:

```suggestion
  return variable.defs.some(definition => {
```

Definitely not a blocker for review. I can make this change since it's the only comment I have 🙂 otherwise this PR looks great!